pkg/stub/handler.go

```diff
@@ -2,96 +2,531 @@ package stub
+				return nil
+			}
+		}
+		secretToCreate := corev1.Secret{}
+		secret.DeepCopyInto(&secretToCreate)
+		secretToCreate.Namespace = ""
+		secretToCreate.ResourceVersion = ""
+		secretToCreate.UID = ""
+
+		s, err := h.secretclientwrapper.Get("openshift", secret.Name)
+		if err != nil && !kerrors.IsNotFound(err) {
+			return h.processError(samplesResource, v1alpha1.SecretUpdateFailed, err, "failed to get registry dockerconfig secret in openshift namespace : %v")
+		}
 		if err != nil {
-			logrus.Errorf("failed to get image client : %v", err)
-			return err
+			s = nil
+		}
+		if s != nil {
+			logrus.Printf("updating dockerconfig secret %s in openshift namespace", v1alpha1.SamplesRegistryCredentials)
+			_, err = h.secretclientwrapper.Update("openshift", &secretToCreate)
+		} else {
+			logrus.Printf("creating dockerconfig secret %s in openshift namespace", v1alpha1.SamplesRegistryCredentials)
+			_, err = h.secretclientwrapper.Create("openshift", &secretToCreate)
+		}
+		if err != nil {
+			return h.processError(samplesResource, v1alpha1.SecretUpdateFailed, err, "failed to create/update registry dockerconfig secret in openshif namespace : %v")
+		}
+		newStatus = corev1.ConditionTrue
+		h.registrySecret = secret
+	}
+
+	err := h.GoodStatusUpdate(samplesResource, newStatus, v1alpha1.ImportCredentialsExist)
+	if err != nil {
+		return err
+	}
+
+	h.registrySecret = secret
+	h.samplesResource = samplesResource
+
+	return nil
+}
+
+func (h *Handler) Handle(ctx context.Context, event sdk.Event) error {
+	switch event.Object.(type) {
+	case *corev1.Secret:
+		dockercfgSecret, _ := event.Object.(*corev1.Secret)
+
+		if dockercfgSecret.Name != v1alpha1.SamplesRegistryCredentials {
+			return nil
 		}
-		logrus.Printf("image client %#v", imageclient)
-		templist, err := tempclient.Templates("openshift").List(metav1.ListOptions{})
-		logrus.Printf("template list %#v", templist)
-		/*err := sdk.Create(newbusyBoxPod(o))
-		if err != nil && !errors.IsAlreadyExists(err) {
-			logrus.Errorf("failed to create busybox pod : %v", err)
+
+		if h.samplesResource == nil {
+			if !event.Deleted {
+				h.registrySecret = dockercfgSecret
+			}
+			return nil
+		}
+
+		err := h.manageDockerCfgSecret(event.Deleted, h.samplesResource, dockercfgSecret)
+		if err != nil {
 			return err
-		}*/
+		}
+
+	case *v1alpha1.SamplesResource:
+		newStatus := corev1.ConditionTrue
+		var srcfg *v1alpha1.SamplesResource
+		if event.Deleted {
+			logrus.Println("SampleResource deleted")
+			h.samplesResource = nil
+			return h.CreateDefaultResourceIfNeeded()
+		} else {
+			srcfg, _ = event.Object.(*v1alpha1.SamplesResource)
+		}
+
+		// coordinate with timer's check on creating
+		// default resource ... looks at h.sampleResource,
+		// which is not set until this whole case is completed
+		h.mutex.Lock()
+		defer h.mutex.Unlock()
+
+		if h.samplesResource != nil &&
+			// in case a new, non-default resource is created
+			h.samplesResource.Name == srcfg.Name {
```

Comment:
i don't know what we'd do if someone did create a second samplesresource instance?

Reply:
Yeah similar to the secret name, I was wondering if we should pin down via constant the name of the samplesresource we deal with, and document that is the only name we consider.  Just stopped short of implementing it, and decided this minimal step forward was enough to generate the discussion.

Thoughts on the static, well defined name ?

Reply:
@jwforres is there any current thinking around how "singleton" operators are supposed to deal with this, or even per-namespace operators that can still only handle a single instance (per namespace) of the thing they instantiate?

What are we supposed to do if someone creates 2+ CRs?  Can our operator simply say "we only look at the CR named "default" and ignore all the others, and you must name your CR instance "foo"?  Is there some other pattern that's been established?

Reply:
Using a static name is one option, the other one i've heard discussed is marking the first one you see with an annotation or something and that becomes the one the operator is managing. I can see downsides to both. But the annotation method seems like it would be easier for an admin to screw things up and the operator to not know which one it is supposed to be managing, although since you in theory have to be able to take a CR in and out of operator management anyway, it should be resolvable if they do mess it up.

Reply:
static name it is.

As for the name, how about "openshift-samples"?